text/0080-datetime-extension.md

```diff
@@ -150,14 +155,14 @@ Values of type `datetime` have the following methods:
 - `.toDate()` returns a new `datetime`, truncating to the day, such that printing the `datetime` would have `00:00:00` as the time.
 - `.toTime()` returns a new `duration`, removing the days, such that only milliseconds since `.toDate()` are left. This is equivalent to `DT.durationSince(DT.toDate())`
 
-Values of type `datetime` can be used with comparison operators:
+Values of type `datetime` can be compared using the following functions:
+
+- `DT1.before(DT2)` returns `true` when `DT1` is before `DT2`
```

Comment:
I liked operator overloading, what is the justification for removing it? I think putting both of those in one PR is going to muddle things?

Reply:
+1 to Aaron's comment.  We shouldn't have two operators that do the same thing.

Reply:
IIRC from the meeting, we sort of agreed to implement RFC 80 without operator overloading and then come back and add overloading for both datetime and decimal. At least personally, I don't think doing that requires amending the RFC because the end goal remains unchanged.

Reply:
I just wanted to call out that the implementation will not do operator overloading. But rather another and later PR will do it.

Reply:
I'm fine with not supporting overloading (now or ever).

But in that case, we should amend this RFC to take out all mentions overloading in the main body and in examples, and maybe add it as an `Alternative` at the bottom.

This is important because we'll also need to write documentation for Datetime, and in the docs, we have to commit to some way of representing comparison.

If overloading is coming at some undefined later point, I think it merits a separate RFC.

Reply:
The updated version removes operator overloading, which will be described in its own RFC.

Reply:
I'm saying leave operator overloading in the RFC. (As strongdm has already implemented it that way, and we want that way to be standard.)

Reply:
I agree with @aaronjeline.  We should leave operator overloading in the RFC.  That doesn't mean it has to be in the same _PR_ that the other datetime operations are added in; for all I care, we could have a separate PR per datetime operation if the implementer thought that made sense organizationally.  But when we stabilize the datetime feature and consider the RFC complete, it should include operator overloading.  (Among other reasons, because StrongDM has done it that way in the Go implementation.)

Reply:
If StrongDM has already implemented overloading, then I agree with should do the same.

My main objection is to having _both_ overloading and methods.  I think that's confusing and unnecessary.  I would have been ok with either, but existing implementation takes precedence.

On a related note, we should write a separate RFC for supporting overloading for Decimals and deprecating / removing the comparison methods.  As a user, I'd find it weird / counterintuitive that I can compare Dates with operators but not Decimals.

Reply:
To sum up, there is no clear winner between operator overloading and non-overloading, as indicated by the table below.

```
|             | overloading                   | non-overloading |
|-------------|-------------------------------|-----------------|
| consistency | N (or Y with breaking change) | Y               |
| elegancy    | Y                             | N               |
```

To make the implementation both consistent and elegant, we should make breaking changes to the `decimal` extension, which will be another RFC, as Emina pointed out. If that's the case, I don't think it hurts to disallow operator overloading for this RFC and propose a new RFC to allow it for all extensions.

Reply:
Again, I disagree. Leaving aside decimal for a second, we already agreed to support operator overloading, and we want to support it eventually, and someone else has already implemented it based on that decision, so why remove it from the RFC?
If we think having both overloaded operators and methods is confusing duplication, then we shouldn't add methods.

Reply:
I agree with Aaron on this. We accepted the RFC as written, an implementation exists according to that spec, and we intended to support it as written eventually, so removing it from the RFC is silly.

It's fine to merge a partial implementation of the RFC while creating an issue to finish the implementation.